core/src/main/java/org/mule/runtime/core/api/MuleContext.java

```diff
@@ -382,5 +382,10 @@ default StreamingManager getStreamingManager() {
    */
   Properties getDeploymentProperties();
 
+
+  /**
+   * Sets the log level at which the Splash Screen will be logged to debug.
+   */
+  default void setDebugSplashScreenLevel() {}
```

Comment:
I think we can avoid adding this to API considering we are only going to be using it internally, and that it will only have any effect on `DefaultMuleContext` instances. I would prefer not to be stuck with this forever if we can avoid it.

Technically, we could just put it in `DefaultMuleContext` and do the `instanceof` check on `DefaultApplicationPolicyInstance`. Neither `org.mule.runtime.core` nor `org.mule.runtime.deployment.model.impl` modules should go to the API module layer anyway. See the [Multi Mules ADR](https://docs.google.com/document/d/1Js0hauEpnAEFInOJXrvROkjlALCe5LaHJ2YcHu6DqRE) for more information (or the [APIs split BOM](https://github.com/mulesoft/mule-ee-distributions/blob/master/bom/runtime-apis-split-loader-bom/pom.xml))

But the latest convention is to be more explicit regarding things that are inter-module APIs, so **we should make it privileged and put it in `PrivilegedMuleContext`**.

Also, we would need to add the `@since` in the JDoc (adjusted to the version you will CP this to, which I believe will be 4.4).